include/ncengine/graphics/MeshRenderer.h

```diff
@@ -3,68 +3,87 @@
 #include "ncengine/asset/Assets.h"
 #include "ncengine/ecs/Component.h"
 
+#include <memory>
+
 namespace nc::graphics
 {
-    struct PbrMaterial
-    {
-        std::string baseColor;
-        std::string normal;
-        std::string roughness;
-        std::string metallic;
-
-        #ifdef NC_EDITOR_ENABLED
-        void EditorGuiElement();
-        #endif
-    };
-
-    enum class TechniqueType : uint8_t
-    {
-        None = 0,
-        PhongAndUi = 1
-    };
-
-    struct PbrMaterialView
-    {
-        TextureView baseColor;
-        TextureView normal;
-        TextureView roughness;
-        TextureView metallic;
-    };
-
-    class MeshRenderer : public ComponentBase
-    {
-        NC_ENABLE_IN_EDITOR(MeshRenderer)
-
-        public:
-            MeshRenderer(Entity entity,
-                         std::string meshUid,
-                         PbrMaterial material,
-                         TechniqueType techniqueType = TechniqueType::PhongAndUi);
-
-            auto GetMesh() const -> const MeshView& { return m_mesh; }
-            auto GetMaterialView() const -> const PbrMaterialView& { return m_materialView; }
-            auto GetTechniqueType() const -> TechniqueType { return m_techniqueType; }
-
-            void SetMesh(std::string meshUid);
-            void SetBaseColor(const std::string& texturePath);
-            void SetNormal(const std::string& texturePath);
-            void SetRoughness(const std::string& texturePath);
-            void SetMetallic(const std::string& texturePath);
-
-            #ifdef NC_EDITOR_ENABLED
-            auto GetMaterial() -> PbrMaterial& { return m_material; }
-            auto GetMeshPath() const -> const std::string& { return m_meshPath; }
-            #endif
-
-        private:
-            #ifdef NC_EDITOR_ENABLED
-            PbrMaterial m_material;
-            std::string m_meshPath;
-            #endif
-            MeshView m_mesh;
-            PbrMaterialView m_materialView;
-            TechniqueType m_techniqueType;
-    };
+struct PbrMaterial
+{
+    std::string baseColor;
+    std::string normal;
+    std::string roughness;
+    std::string metallic;
+};
+
+struct PbrMaterialView
+{
+    TextureView baseColor;
+    TextureView normal;
+    TextureView roughness;
+    TextureView metallic;
+};
+
+enum class TechniqueType : uint8_t
+{
+    None = 0,
+    PhongAndUi = 1
+};
+
+class MeshRenderer : public ComponentBase
+{
+    NC_ENABLE_IN_EDITOR(MeshRenderer)
+
+    public:
+        MeshRenderer(Entity entity,
+                     std::string meshUid,
+                     PbrMaterial material,
+                     TechniqueType techniqueType = TechniqueType::PhongAndUi);
+
+        auto GetMeshView() const noexcept -> const MeshView&
```

Comment:
Why expand these blocks?

Reply:
Can't say I thought too much about it - I added `noexcept` to them and they felt a little long. I could be talked into changing them back.